src/components/Controls/TableRow.vue

```diff
@@ -5,28 +5,77 @@
     :class="readOnly ? '' : 'hover:bg-gray-25 dark:hover:bg-gray-900'"
   >
     <!-- Index or Remove button -->
-    <div class="flex items-center ps-2 text-gray-600 dark:text-gray-400">
-      <span class="hidden" :class="{ 'group-hover:inline-block': !readOnly }">
+    <div
+      class="flex items-center ps-2 text-gray-600 dark:text-gray-400"
+      @mouseenter="isRowIndexVisible = false"
+      @mouseleave="isRowIndexVisible = true"
+    >
+      <span class="relative w-4 h-4 flex items-center justify-center">
         <feather-icon
+          v-if="!readOnly && !isRowIndexVisible"
           name="x"
-          class="w-4 h-4 -ms-1 cursor-pointer"
+          class="
+            w-4
+            h-4
+            -ms-1
+            cursor-pointer
+            rounded
+            focus:outline-none focus:ring-2 focus:ring-blue-500 focus:bg-blue-50
+            dark:focus:bg-gray-800
+            transition
+          "
           :button="true"
+          tabindex="0"
+          role="button"
+          aria-label="Delete row"
           @click="$emit('remove')"
+          @keydown.enter="$emit('remove')"
+          @keydown.space.prevent="$emit('remove')"
+          @keydown.shift.tab.prevent="focusPreviousElement"
+          @focus="isDeleteFocused = true"
+          @blur="isDeleteFocused = false"
         />
+        <span
+          v-if="!readOnly && !isDeleteFocused && isRowIndexVisible"
+          class="
+            absolute
+            left-0
+            top-0
+            w-full
+            h-full
+            flex
+            items-center
+            justify-center
+            focus:outline-none
+            focus-visible:ring-2 focus-visible:ring-blue-500
+            rounded
+          "
+          tabindex="0"
+          role="button"
+          aria-label="Delete row"
+          @focus="handleShiftTabFocus"
+          @keydown.shift.tab.prevent="focusPreviousElement"
+          @keydown.enter="$emit('remove')"
+          @keydown.space.prevent="$emit('remove')"
```

Comment:
Unnecessary event bindings

```suggestion
          @keydown.enter="$emit('remove')"
```